collector.c

```diff
@@ -163,10 +163,10 @@ probe_waits(History *observations, HTAB *profile_hash,
 					   *observation;
 		PGPROC		   *proc = &ProcGlobal->allProcs[i];
 
-		if (proc->pid == 0)
+		if (proc->wait_event_info == 0 && !pgws_collector_hdr->sampleCpu)
 			continue;
 
-		if (proc->wait_event_info == 0)
+		if (proc->pid == 0 || proc->procLatch.owner_pid == 0 || proc->pid == MyProcPid)
```

Comment:
Please add a comment here explaining why we mess with the latch.

Reply:
Comment added.